WordPress/Classes/Services/CommentService.m

```diff
@@ -104,6 +107,50 @@ - (void)syncCommentsForBlog:(Blog *)blog
                        }];
 }
 
+- (void)loadMoreCommentsForBlog:(Blog *)blog
+                        success:(void (^)(BOOL hasMore))success
+                        failure:(void (^)(NSError *))failure
+{
+    id<CommentServiceRemote> remote = [self remoteForBlog:blog];
+    NSMutableDictionary *options = [NSMutableDictionary dictionary];
+    if ([remote isKindOfClass:[CommentServiceRemoteREST class]]) {
+        NSString *entityName = NSStringFromClass([Comment class]);
+        NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:entityName];
+        request.predicate = [NSPredicate predicateWithFormat:@"dateCreated != NULL && blog=%@", blog];
+        NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"dateCreated" ascending:YES];
+        request.sortDescriptors = @[sortDescriptor];
+        Comment *oldestComment = [[self.managedObjectContext executeFetchRequest:request error:nil] firstObject];
+        if (oldestComment.dateCreated) {
+            options[@"before"] = [oldestComment.dateCreated WordPressComJSONString];
+            options[@"order"] = @"desc";
+        }
+    } else if ([remote isKindOfClass:[CommentServiceRemoteXMLRPC class]]) {
+        NSUInteger postCount = [blog.posts count];
+        options[@"offset"] = @(postCount);
+    }
+    [remote getCommentsForBlog:blog
+        options:options
+        success:^(NSArray *comments) {
+                       [self.managedObjectContext performBlock:^{
+                           [self mergeComments:comments
+                                       forBlog:blog
+                                 purgeExisting:NO
+                             completionHandler:^{
+                                 if (success) {
+                                     success(comments.count > 1);
+                                 }
+                             }];
+                       }];
+        }
+        failure:^(NSError *error) {
+                       if (failure) {
+                           [self.managedObjectContext performBlock:^{
```

Comment:
Is the `performBlock` needed here?  If not, maybe just pass the `failure` block here.